pandas/core/frame.py

```diff
@@ -2708,7 +2727,7 @@ def info(
             show_counts=show_counts,
         )
 
-    def memory_usage(self, index=True, deep=False) -> Series:
+    def memory_usage(self, index: bool = True, deep: bool = False) -> Series:
```

Comment:
return type is `self._constructor_sliced`. The only way to tie the result type to a class attribute is to make DataFrame/Series generic. out of scope for now. see #32638

Reply:
Ok - reverted